crates/starknet-rs-py/src/types/block_info.rs

```diff
@@ -0,0 +1,83 @@
+use cairo_felt::Felt;
+use num_bigint::BigUint;
+use pyo3::{exceptions::PyRuntimeError, prelude::*};
+use starknet_rs::{business_logic::state::state_api_objects::BlockInfo, utils::Address};
+
+#[pyclass]
+#[pyo3(name = "BlockInfo")]
+#[derive(Debug)]
+pub struct PyBlockInfo {
+    inner: BlockInfo,
+}
+
+#[pymethods]
+impl PyBlockInfo {
+    #[new]
+    fn new(
+        block_number: u64,
+        block_timestamp: u64,
+        gas_price: u64,
+        sequencer_address: BigUint,
+        starknet_version: String,
+    ) -> Self {
+        let address = Address(Felt::from(sequencer_address));
+        let inner = BlockInfo {
+            block_number,
+            block_timestamp,
+            gas_price,
+            sequencer_address: address,
+            starknet_version,
+        };
+        Self { inner }
+    }
+
+    /// Returns an empty BlockInfo object; i.e., the one before the first in the chain.
+    #[staticmethod]
+    fn empty(address: BigUint) -> PyBlockInfo {
+        let inner = BlockInfo::empty(Address(Felt::from(address)));
+        Self { inner }
+    }
+
+    /// Returns a BlockInfo object with default gas_price.
+    #[staticmethod]
```

Comment:
Careful, `@staticmethod` is not the same as `@classmethod`. A static method may only be called from the class type, but a `@classmethod` can be called from both the class type and any instance.

In our case, I don't think anything will happen if we leave them as a `#[staticmethod]`, but just in case I'd replace them with `#[classmethod]`.

Reply:
Oh, I thought they were the same. Well, good to know! I'll change it in a bit

Reply:
Changed!

Reply:
In this case we want a static method since empty and the other function are constructor so they are called using the class.

`BlockInfo.empty()`

Reply:
The difference between static and class methods in Python is that the second one receives the class as an implicit parameter and apart from that, both are used in the same way: `BlockInfo.empty()`. I believe these are used for easy subclassing because they allow constructors to return an object of the class received (which could be a subclass), instead of from the base class.